## 목차
1. 필요 라이브러리 불러오기<br><br>

2. 데이터셋 불러오기<br><br>

3. 데이터 확인<br>
    3-1. 형태, features, 크기 파악<br><br>
    
4. 데이터 전처리<br>
    4-1. 각 feature 파악/탐색<br>
    4-2. 불필요한 feature 삭제<br>
    4-3. feature 결합<br>
    4-4. 전처리 데이터 확인<br><br>
    
5. 요약

## 1. 필요 라이브러리 불러오기

In [1]:
import pandas as pd
import numpy as np

+ 하기 사용 csv 파일 엑셀로 열었을 시, 글자가 깨지는 현상 발생  
+ 원인: 한글 인코딩 후, 디코딩 시 다른 규격 적용
    - eg) csv 파일 생성 시, utf-8 규격 인코딩 진행 -> 엑셀 상 csv 열 때, 949: 한국어 ANSI 코드 디코딩 설정 -> 한글 깨짐  
+ 해결:  
    1. 윈도우 메모장 사용  
    2. 메모장에서 해당 cvs 열기  
    3. 다른 이름으로 저장; 하단 인코딩: ANSI 선택  
    4. 저장  

## 2. 데이터셋 불러오기

In [2]:
comm = pd.read_csv('C:/Python/01 Group Works/Project_InnoPost/Received Data/07_상권정보(대전)/소상공인시장진흥공단_상가(상권)정보_대전_202109.csv', encoding = 'cp949')

# UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbb in position 1: invalid start byte --> 한글 포함 csv 혹은 txt 파일 포함 시 해당 오류 발생 --> encoding = 'cp494' (euc-kr의 확장 버전) 혹은 'euc-kr' 사용하여 인코딩 후 출력 

## 3. 데이터 확인

In [3]:
# pd.options.display.max_rows = 6
    # 추후 보여질 데이터프레임의 열 개수 지정

In [4]:
comm

상가업소번호         상호명    지점명 상권업종대분류코드 상권업종대분류명 상권업종중분류코드   상권업종중분류명  \
0      22401931      스피드메이트    NaN         F    생활서비스       F14    자동차/이륜차   
1      24185362  국제종합기계부품총판    NaN         D       소매       D25      기타판매업   
2      22742209    알파수학전문학원    NaN         R    학문/교육       R01  학원-보습교습입시   
3      17174039         솔바람    NaN         Q       음식       Q01         한식   
4      17174056       BBQ치킨  대전대흥점         Q       음식       Q09       유흥주점   
...         ...         ...    ...       ...      ...       ...        ...   
69578  17715966         공부방    엘이네         R    학문/교육       R09       학원기타   
69579  17699707        어썸스파    NaN         F    생활서비스       F01    이/미용/건강   
69580  17720100         쌀통닭    갈마점         Q       음식       Q05     닭/오리요리   
69581  17709625    202사고력수학    NaN         R    학문/교육       R09       학원기타   
69582  11804773         BE밥    NaN         Q       음식       Q01         한식   

      상권업종소분류코드   상권업종소분류명 표준산업분류코드  ...                     건물관리번호  \
0        F14A01  자동차정비/카센타      NaN  ...  3020014600111220000001772   
1        D25A15      농기구판매   G47512  ...  3017010300101730001030969   
2        R01A01      학원-입시   P85501  ...  3014010200101570008028501   
3        Q01A01  한식/백반/한정식   I56111  ...  3020012000109080016009529   
4        Q09A01      호프/맥주   I56219  ...  3014010500101180005012316   
...         ...        ...      ...  ...                        ...   
69578    R09A11      학원-기타   P85699  ...  3023011100104830000022549   
69579    F01A03    비만/피부관리      NaN  ...  3017010600100910023000001   
69580    Q05A08  후라이드/양념치킨   I56193  ...  3017011100104270083011985   
69581    R09A11      학원-기타   P85699  ...  3017011200109080000018415   
69582    Q01A01  한식/백반/한정식   I56111  ...  3014010500104640001013967   

                 건물명                    도로명주소   구우편번호  신우편번호  동정보  층정보  호정보  \
0                NaN        대전광역시 유성구 관들4길 25  305509  34018  NaN  NaN  NaN   
1               써머스빌        대전광역시 서구 계백로 1426  302160  35400  NaN  NaN  NaN   
2               선우빌딩      대전광역시 중구 선화서로 128-1  301824  34841  NaN    2  NaN   
3                NaN  대전광역시 유성구 은구비서로3번길 8-23  305330  34086  NaN  NaN  NaN   
4                NaN         대전광역시 중구 대흥로 158  301802  34928  NaN    1  NaN   
...              ...                      ...     ...    ...  ...  ...  ...   
69578  금강센트럴파크서희스타힐스  대전광역시 대덕구 대덕대로1585번길 44  306190  34305  110  NaN  803   
69579      한밭새마을금고사옥         대전광역시 서구 계룡로 599  302857  35264  NaN    2  205   
69580            NaN         대전광역시 서구 신갈마로 61  302808  35280  NaN    1  104   
69581      샘머리아파트2단지         대전광역시 서구 청사로 281  302827  35205  202  NaN  101   
69582            NaN     대전광역시 중구 중앙로112번길 42  301803  34916  NaN  NaN  NaN   

               경도         위도  
0      127.393241  36.424448  
1      127.379765  36.312625  
2      127.413968  36.334319  
3      127.320105  36.373504  
4      127.429579  36.324517  
...           ...        ...  
69578  127.425727  36.454575  
69579  127.390199  36.341275  
69580  127.366360  36.339853  
69581  127.392011  36.361815  
69582  127.423312  36.325550  

[69583 rows x 39 columns]

In [5]:
comm.info()
    # 데이터셋 정보 확인

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69583 entries, 0 to 69582
Data columns (total 39 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   상가업소번호     69583 non-null  int64  
 1   상호명        69583 non-null  object 
 2   지점명        11158 non-null  object 
 3   상권업종대분류코드  69583 non-null  object 
 4   상권업종대분류명   69583 non-null  object 
 5   상권업종중분류코드  69583 non-null  object 
 6   상권업종중분류명   69583 non-null  object 
 7   상권업종소분류코드  69583 non-null  object 
 8   상권업종소분류명   69583 non-null  object 
 9   표준산업분류코드   64985 non-null  object 
 10  표준산업분류명    64985 non-null  object 
 11  시도코드       69583 non-null  int64  
 12  시도명        69583 non-null  object 
 13  시군구코드      69583 non-null  int64  
 14  시군구명       69583 non-null  object 
 15  행정동코드      69583 non-null  int64  
 16  행정동명       69583 non-null  object 
 17  법정동코드      69583 non-null  int64  
 18  법정동명       69583 non-null  object 
 19  지번코드       69583 non-null  int64  
 20  대지구분코드

In [6]:
%%html
<style>
table {float: left}
</style>

## 4. 데이터 전처리
### 4-1. 각 feature 탐색

| #   	| COL NAME           	| USE 	| NOTES         	|
|:-----	|:--------------------	|:-----:|:---------------	|
| 1.  	| 상가업소번호       	| X   	|               	|
| 2.  	| 상호명             	|     	| 지점명과 결합; 상호명만 명시할 시, 타 지역 상권과 혼동 가능성 고려|
| 3.  	| 지점명             	|     	| 상단 참고     	|
| 4.  	| 상권업종대분류코드 	| X   	|               	|
| 5.  	| 상권업종대분류명   	|     	|흡연지수 산출 목적|
| 6.  	| 상권업종중분류코드 	| X   	|               	|
| 7.  	| 상권업종중분류명   	|     	|흡연지수 산출 목적|
| 8.  	| 상권업종소분류코드 	| X   	|               	|
| 9.  	| 상권업종소분류명   	|     	|흡연지수 산출 목적|
| 10. 	| 표준산업분류코드   	| X   	|               	|
| 11. 	| 표준산업분류명     	| X    	|               	|
| 12. 	| 시도코드           	| X   	|               	|
| 13. 	| 시도명             	|     	|지역단위 별 상권수 등 계산 목적|
| 14. 	| 시군구코드         	| X   	|               	|
| 15. 	| 시군구명           	|     	|지역단위 별 상권수 등 계산 목적|
| 16. 	| 행정동코드         	| X   	|               	|
| 17. 	| 행정동명           	|     	|지역단위 별 상권수 등 계산 목적|
| 18. 	| 법정동코드         	| X   	|               	|
| 19. 	| 법정동명           	| X   	|               	|
| 20. 	| 지번코드           	| X   	|               	|
| 21. 	| 대지구분코드       	| X   	|               	|
| 22. 	| 대지구분명         	| X   	|               	|
| 23. 	| 지번본번지         	| X   	|               	|
| 24. 	| 지번부번지         	| X   	|               	|
| 25. 	| 지번주소           	|X     	|               	|
| 26. 	| 도로명코드         	| X   	|               	|
| 27. 	| 도로명             	| X   	|               	|
| 28. 	| 건물본번지         	| X   	|               	|
| 29. 	| 건물부번지         	| X   	|               	|
| 30. 	| 건물관리번호       	| X   	|               	|
| 31. 	| 건물명             	| X   	|               	|
| 32. 	| 도로명주소         	|     	|각 상권의 상세주소로서 참고|
| 33. 	| 구우편번호         	| X   	|               	|
| 34. 	| 신우편번호         	| X   	|               	|
| 35. 	| 동정보             	| X   	|               	|
| 36. 	| 층정보             	| X   	|               	|
| 37. 	| 호정보             	| X   	|               	|
| 38. 	| 경도               	|     	| 시각화 목적|
| 39. 	| 위도               	|     	| 시각화 목적|

총 39개 열 중,
- 흡연지수 도출 및 QGIS 시각화 시, 불필요한 열 삭제 예정; (28)
    1. 각 상권, 지역에 부여된 식별 번호/코드 + 표준산업분류명 (6) 
    2. 지역/주소관련 코드 및 상세/중복 정보 삭제 (22)<br><br>
- 사용 가능 정보; (11)
    1. 상호명 + 지점명 (2)
    2. 상권업종대분류명
    3. 상권업종중분류명
    4. 상권업종소분류명
    5. 시도명
    6. 시군구명
    7. 행정동명
    8. 도로명주소
    9. 경도
    10. 위도

### 4-2. 불필요한 feature 삭제

In [7]:
comm.drop(['상가업소번호', '상권업종대분류코드', '상권업종중분류코드', '상권업종소분류코드', '표준산업분류코드', '표준산업분류명', '시도코드', '시군구코드', '행정동코드', '법정동코드', '법정동명', '지번코드', '대지구분코드', '대지구분명', '지번본번지', '지번부번지', '지번주소', '도로명코드', '도로명', '건물본번지', '건물부번지', '건물관리번호', '건물명', '구우편번호', '신우편번호', '동정보', '층정보', '호정보'], axis = 'columns', inplace = True)
    # 열 삭제 진행
    # axis = 'columns' == axis = 1; 열 단위 삭제
    # inplace = True; 변경사항 저장

### 4-3. feature 결합

In [8]:
comm['지점명'].fillna('', inplace = True)
    # NaN --> 공란 변경, 저장

comm['상호+지점명'] = comm['상호명'] + ' ' + comm['지점명']
    # 상호명 + 지점명 결합 --> 새로운 열, 상호+지점명 생성

In [9]:
comm.drop(['상호명', '지점명'], axis = 1, inplace = True)

In [10]:
comm = comm[['상호+지점명', '상권업종대분류명', '상권업종중분류명', '상권업종소분류명', '시도명', '시군구명', '행정동명', '도로명주소', '경도', '위도']]

In [11]:
comm

상호+지점명 상권업종대분류명   상권업종중분류명   상권업종소분류명    시도명 시군구명   행정동명  \
0          스피드메이트     생활서비스    자동차/이륜차  자동차정비/카센타  대전광역시  유성구    관평동   
1      국제종합기계부품총판        소매      기타판매업      농기구판매  대전광역시   서구   도마2동   
2        알파수학전문학원     학문/교육  학원-보습교습입시      학원-입시  대전광역시   중구  은행선화동   
3             솔바람        음식         한식  한식/백반/한정식  대전광역시  유성구   노은1동   
4      BBQ치킨 대전대흥점       음식       유흥주점      호프/맥주  대전광역시   중구    대흥동   
...            ...      ...        ...        ...    ...  ...    ...   
69578      공부방 엘이네    학문/교육       학원기타      학원-기타  대전광역시  대덕구    석봉동   
69579        어썸스파     생활서비스    이/미용/건강    비만/피부관리  대전광역시   서구    탄방동   
69580      쌀통닭 갈마점       음식     닭/오리요리  후라이드/양념치킨  대전광역시   서구   갈마1동   
69581    202사고력수학     학문/교육       학원기타      학원-기타  대전광역시   서구   둔산2동   
69582         BE밥        음식         한식  한식/백반/한정식  대전광역시   중구    대흥동   

                         도로명주소          경도         위도  
0            대전광역시 유성구 관들4길 25  127.393241  36.424448  
1            대전광역시 서구 계백로 1426  127.379765  36.312625  
2          대전광역시 중구 선화서로 128-1  127.413968  36.334319  
3      대전광역시 유성구 은구비서로3번길 8-23  127.320105  36.373504  
4             대전광역시 중구 대흥로 158  127.429579  36.324517  
...                        ...         ...        ...  
69578  대전광역시 대덕구 대덕대로1585번길 44  127.425727  36.454575  
69579         대전광역시 서구 계룡로 599  127.390199  36.341275  
69580         대전광역시 서구 신갈마로 61  127.366360  36.339853  
69581         대전광역시 서구 청사로 281  127.392011  36.361815  
69582     대전광역시 중구 중앙로112번길 42  127.423312  36.325550  

[69583 rows x 10 columns]

### 4-4. 전처리 데이터 확인

In [12]:
comm.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69583 entries, 0 to 69582
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   상호+지점명    69583 non-null  object 
 1   상권업종대분류명  69583 non-null  object 
 2   상권업종중분류명  69583 non-null  object 
 3   상권업종소분류명  69583 non-null  object 
 4   시도명       69583 non-null  object 
 5   시군구명      69583 non-null  object 
 6   행정동명      69583 non-null  object 
 7   도로명주소     69583 non-null  object 
 8   경도        69583 non-null  float64
 9   위도        69583 non-null  float64
dtypes: float64(2), object(8)
memory usage: 5.3+ MB


In [13]:
comm[['상호+지점명', '상권업종대분류명', '상권업종중분류명', '상권업종소분류명', '시도명', '시군구명', '행정동명', '도로명주소']].describe()

상호+지점명 상권업종대분류명 상권업종중분류명   상권업종소분류명    시도명   시군구명   행정동명  \
count   69583    69583    69583      69583  69583  69583  69583   
unique  62120        8       88        574      1      5     79   
top      부동산        소매       한식  한식/백반/한정식  대전광역시     서구   온천1동   
freq       41    23353     8916       5984  69583  22465   3199   

                   도로명주소  
count              69583  
unique             29398  
top     대전광역시 서구 계룡로 598  
freq                 447

In [14]:
pd.set_option('display.max_rows', None)

comm_mod = pd.pivot_table(comm,
                          index = ['상권업종대분류명', '상권업종중분류명'],
                          columns = '시군구명',
                          values = '상호+지점명',
                          aggfunc = 'count')
    # 피벗테이블 생성

comm_mod

시군구명                       대덕구      동구      서구     유성구      중구
상권업종대분류명 상권업종중분류명                                             
관광/여가/오락 PC/오락/당구/볼링등     70.0    82.0   155.0   122.0    63.0
         경마/경륜/성인오락        2.0     2.0     5.0     4.0     2.0
         놀이/여가/취미          2.0     3.0     2.0     3.0     3.0
         무도/유흥/가무        127.0    98.0   220.0   129.0   132.0
         스포츠/운동            2.0     1.0     2.0     4.0     2.0
         연극/영화/극장          2.0     1.0    10.0    15.0     8.0
         요가/단전/마사지        30.0    24.0   136.0    95.0    50.0
부동산      부동산중개           213.0   301.0   770.0   683.0   317.0
         분양                6.0     6.0    22.0    14.0     6.0
         평가/개발/관리          NaN     1.0    11.0     5.0     2.0
생활서비스    개인/가정용품수리        81.0    90.0   259.0   121.0   110.0
         개인서비스             9.0     9.0    46.0    25.0    30.0
         광고/인쇄            59.0   237.0   163.0    80.0   109.0
         기타서비스업           23.0    13.0    82.0    86.0    32.0
         대중목욕탕/휴게         32.0    27.0    46.0    27.0    23.0
         대행업               7.0    21.0    25.0    22.0    10.0
         물품기기대여           17.0    28.0    69.0    30.0    30.0
         법무세무회계            NaN     NaN    35.0     1.0     1.0
         사진               22.0    37.0   134.0    61.0    45.0
         세탁/가사서비스        256.0   261.0   577.0   355.0   312.0
         예식/의례/관혼상제        9.0     3.0    58.0    18.0    15.0
         운송/배달/택배         33.0    28.0    47.0    26.0    42.0
         이/미용/건강         667.0   699.0  2071.0  1222.0   727.0
         인력/고용/용역알선       27.0    33.0    56.0    26.0    48.0
         자동차/이륜차         356.0   214.0   359.0   364.0   298.0
         장례/묘지            12.0     7.0    32.0     6.0    19.0
         주유소/충전소          56.0    25.0    46.0    50.0    28.0
         주택수리            104.0    67.0   154.0    52.0    86.0
         행사/이벤트           18.0    13.0    51.0    24.0    30.0
소매       가구소매             76.0    60.0   111.0    79.0    95.0
         가방/신발/액세서리       88.0   194.0   304.0   129.0   168.0
         가전제품소매           42.0    65.0   143.0    75.0    78.0
         가정/주방/인테리어      494.0   507.0   654.0   420.0   369.0
         건강/미용식품         137.0   156.0   286.0   130.0   176.0
         기타판매업            79.0    58.0    94.0    71.0    76.0
         사무/문구/컴퓨터       158.0   128.0   298.0   184.0   132.0
         사진/광학/정밀기기소매    147.0   200.0   424.0   262.0   190.0
         선물/팬시/기념품        84.0    91.0   227.0   154.0   148.0
         시계/귀금속소매         26.0    74.0    89.0    38.0    34.0
         애견/애완/동물         15.0    27.0    67.0    37.0    26.0
         예술품/골동품/수석/분재    14.0    15.0    27.0    35.0    19.0
         운동/경기용품소매        56.0    71.0   152.0   108.0    69.0
         유아용품              2.0     8.0    23.0    24.0    11.0
         음/식료품소매         640.0   445.0   711.0   516.0   404.0
         의복의류            332.0   633.0  1136.0   607.0   530.0
         의약/의료품소매         67.0    87.0   104.0    58.0    78.0
         자동차/자동차용품       180.0    95.0   206.0   171.0   142.0
         종교용품판매            1.0     2.0     2.0     3.0     5.0
         종합소매점           598.0   752.0  1199.0   826.0   697.0
         중고품소매/교환         68.0    46.0    58.0    48.0    67.0
         책/서적/도서          33.0    39.0    88.0    57.0    61.0
         철물/난방/건설자재소매    185.0   142.0   232.0   130.0   141.0
         취미/오락관련소매        20.0    20.0    78.0    16.0    34.0
         페인트/유리제품소매       40.0    28.0    48.0    23.0    45.0
         화장품소매           147.0   117.0   395.0   224.0   188.0
숙박       모텔/여관/여인숙        80.0   121.0    46.0    72.0   157.0
         민박/하숙             1.0     NaN     8.0     3.0     2.0
         유스호스텔             NaN     NaN     NaN     2.0     NaN
         캠프/별장/펜션          5.0     2.0    14.0     5.0     NaN
         호텔/콘도             1.0     6.0    13.0    29.0    13.0
스포츠      실내운동시설           67.0    58.0   226.0   208.0    74.0
         실외운동시설            3.0 

**주요 흡연 지역 및 [금연지역](https://easylaw.go.kr/CSP/CnpClsMain.laf?popMenu=ov&csmSeq=908&ccfNo=4&cciNo=1&cnpClsNo=1) 관련 상권업종대분류명 선정하여 필터**

In [15]:
comm_mod.query("상권업종대분류명 == ['관광/여가/오락', '생활서비스', '소매', '숙박', '음식', '학문/교육']")
    # 피벗 테이블 상 특정 열 필터

시군구명                       대덕구      동구      서구     유성구      중구
상권업종대분류명 상권업종중분류명                                             
관광/여가/오락 PC/오락/당구/볼링등     70.0    82.0   155.0   122.0    63.0
         경마/경륜/성인오락        2.0     2.0     5.0     4.0     2.0
         놀이/여가/취미          2.0     3.0     2.0     3.0     3.0
         무도/유흥/가무        127.0    98.0   220.0   129.0   132.0
         스포츠/운동            2.0     1.0     2.0     4.0     2.0
         연극/영화/극장          2.0     1.0    10.0    15.0     8.0
         요가/단전/마사지        30.0    24.0   136.0    95.0    50.0
생활서비스    개인/가정용품수리        81.0    90.0   259.0   121.0   110.0
         개인서비스             9.0     9.0    46.0    25.0    30.0
         광고/인쇄            59.0   237.0   163.0    80.0   109.0
         기타서비스업           23.0    13.0    82.0    86.0    32.0
         대중목욕탕/휴게         32.0    27.0    46.0    27.0    23.0
         대행업               7.0    21.0    25.0    22.0    10.0
         물품기기대여           17.0    28.0    69.0    30.0    30.0
         법무세무회계            NaN     NaN    35.0     1.0     1.0
         사진               22.0    37.0   134.0    61.0    45.0
         세탁/가사서비스        256.0   261.0   577.0   355.0   312.0
         예식/의례/관혼상제        9.0     3.0    58.0    18.0    15.0
         운송/배달/택배         33.0    28.0    47.0    26.0    42.0
         이/미용/건강         667.0   699.0  2071.0  1222.0   727.0
         인력/고용/용역알선       27.0    33.0    56.0    26.0    48.0
         자동차/이륜차         356.0   214.0   359.0   364.0   298.0
         장례/묘지            12.0     7.0    32.0     6.0    19.0
         주유소/충전소          56.0    25.0    46.0    50.0    28.0
         주택수리            104.0    67.0   154.0    52.0    86.0
         행사/이벤트           18.0    13.0    51.0    24.0    30.0
소매       가구소매             76.0    60.0   111.0    79.0    95.0
         가방/신발/액세서리       88.0   194.0   304.0   129.0   168.0
         가전제품소매           42.0    65.0   143.0    75.0    78.0
         가정/주방/인테리어      494.0   507.0   654.0   420.0   369.0
         건강/미용식품         137.0   156.0   286.0   130.0   176.0
         기타판매업            79.0    58.0    94.0    71.0    76.0
         사무/문구/컴퓨터       158.0   128.0   298.0   184.0   132.0
         사진/광학/정밀기기소매    147.0   200.0   424.0   262.0   190.0
         선물/팬시/기념품        84.0    91.0   227.0   154.0   148.0
         시계/귀금속소매         26.0    74.0    89.0    38.0    34.0
         애견/애완/동물         15.0    27.0    67.0    37.0    26.0
         예술품/골동품/수석/분재    14.0    15.0    27.0    35.0    19.0
         운동/경기용품소매        56.0    71.0   152.0   108.0    69.0
         유아용품              2.0     8.0    23.0    24.0    11.0
         음/식료품소매         640.0   445.0   711.0   516.0   404.0
         의복의류            332.0   633.0  1136.0   607.0   530.0
         의약/의료품소매         67.0    87.0   104.0    58.0    78.0
         자동차/자동차용품       180.0    95.0   206.0   171.0   142.0
         종교용품판매            1.0     2.0     2.0     3.0     5.0
         종합소매점           598.0   752.0  1199.0   826.0   697.0
         중고품소매/교환         68.0    46.0    58.0    48.0    67.0
         책/서적/도서          33.0    39.0    88.0    57.0    61.0
         철물/난방/건설자재소매    185.0   142.0   232.0   130.0   141.0
         취미/오락관련소매        20.0    20.0    78.0    16.0    34.0
         페인트/유리제품소매       40.0    28.0    48.0    23.0    45.0
         화장품소매           147.0   117.0   395.0   224.0   188.0
숙박       모텔/여관/여인숙        80.0   121.0    46.0    72.0   157.0
         민박/하숙             1.0     NaN     8.0     3.0     2.0
         유스호스텔             NaN     NaN     NaN     2.0     NaN
         캠프/별장/펜션          5.0     2.0    14.0     5.0     NaN
         호텔/콘도             1.0     6.0    13.0    29.0    13.0
음식       기타음식업            31.0    20.0    27.0    39.0    12.0
         닭/오리요리          226.0   259.0   487.0   337.0   224.0
         별식/퓨전요리          66.0    65.0   135.0    82.0    70.0
         부페                9.0    11.0    30.0    22.0     9.0
         분식              272.0 

## 5. 요약


### 데이터 전처리 진행 상황
+ 데이터 축소
    - 69583 rows × 39 columns --> 69583 rows × 10 columns
    - 20.7+ MB --> 5.3+ MB
    - 최종 데이터; string & float type, non-null 값 구성<br><br>
    
+ 데이터 열 별 사용목적 식별 (총 10개 열)
    1. (1) 상호+지점명: count
    2. (3) 상권업종대분류명 / 상권업종중분류명 / 상권업종소분류명: 흡연지수 산출 시 사용
    3. (3) 시도명 / 시군구명 / 행정동명: 지역단위 별 상권수 등 계산 목적
    4. (1) 도로명주소: 각 상권의 상세주소로서 참고
    5. (2) 경도 / 위도: 시각화 목적

### 추가 고려 필요한 사항
+ 흡연 위험 지역 정의
    - 금연지역임에도 흡연(민원 등) 빈번하게 발생하는 지역
    - 흡연 가능 구역이나 흡연 취약 계층 유동인구가 많은 지역 등<br><br>
    
+ 흡연 취약 계층 정의
    - 미성년자 (만 0-19세)
    - 임산부
    - 비흡연 노인 등 
    - **인구 파악 가능여부 확인 필요**<br><br>
    
+ 폐업장 (2022년 자료 다운로드 완료) 등

### 추후 기타 참고 자료
+ [지역 별 건강 프로파일](https://chs.kdca.go.kr/chs/recsRoom/areaResultMain.do#)
    - 건강행태에 대한 설명 - 비흡연자의 공공장소 간접흡연 노출률 (2020); 서구 > 대덕구 > 중구 > 동구 > 유성구<br><br>
    
+ [지역사회 건강관련 요인 데이터베이스](https://chs.kdca.go.kr/chs/recsRoom/dataBaseMain.do)
    - 대분류: 건강행태
    - 중분류: 흡연
    - 소분류: 간접흡연, 흡연, 금연 등<br><br>
    
+ [기사1](https://www.chungnamilbo.co.kr/news/articleView.html?idxno=673179)
    - 대전 도심 '재떨이화' ...간접흡연에 비흡연자 몸살, 흡연공간 부족...번화가 길바닥 꽁초 가득 (2022.08)<br><br>
    
+ [기사2](http://www.daejonilbo.com/news/articleView.html?idxno=1483063)
    - 대전 금연거리/공원 '무용지물'...보건당국 코로나 방역 업무 지원에 단속 한계 (2021.08)<br><br>